Source/ASDisplayNode.mm

```diff
@@ -2685,10 +2743,15 @@ - (void)_insertSubnodeSubviewOrSublayer:(ASDisplayNode *)subnode atIndex:(NSInte
   // where the view and layer can change. We can avoid locking.
 
   // If we can use view API, do. Due to an apple bug, -insertSubview:atIndex: actually wants a LAYER index, which we pass in
+  __instanceLock__.lock();
+  UIView *view = _view;
+  CALayer *layer = _layer;
+  __instanceLock__.unlock();
```

Comment:
As discussed above, since this method is on main (asserted few lines above), we can relax the locking requirement here.